api/v1alpha1/authorization_types.go

```diff
@@ -0,0 +1,55 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package v1alpha1
+
+// Authorization defines the authorization configuration.
+type Authorization struct {
+	// Rules contains all the authorization rules.
+	// If rules contains at least one Allow rule and none of them
+	// matches the action for the request is deny.
+	// If rules contains at least one Deny rule and none of them
+	// matches the action for the request is allow.
+	// If rules contains Allow and Deny,
+	// and the request matches both, `Deny` takes precedence.
+	//
+	// +kubebuilder:validation:MinItems=1
+	Rules []Rule `json:"rules,omitempty"`
+}
+
+// Rule defines the single authorization rule.
+type Rule struct {
+	// ClientSelectors contains the client selector configuration.
+	// All selectors are AND'd together and only if all selector are valid
+	// the Action is performed.
+	//
+	// +kubebuilder:validation:MinItems=1
+	ClientSelectors []ClientSelector `json:"clientSelector,omitempty"`
```

Comment:
thinking out loud, does this need to be a `list` ? , I know I had suggested it earlier 🙈

Reply:
it depends is all items under ClientSelector going to have array or not. Like in case of cidrs (array), this can be just object. However, when like jwt is added it needs to have array as well.

Reply:
all items under `clientSelector` will be an array to implement the OR/ANY case, so we should use an array for `jwt` settings as well